### PR TITLE
PYIC-6943: fix title home link when running with a url path

### PR DESCRIPTION
**Pull request by sam803**
## Proposed changes

This PR https://github.com/govuk-one-login/ipv-core-back/pull/2134 introduced a link to the title that should take the user to the initial home page, but this does not work on the github page since the url include a path

### What changed

Dynamically set the url for the title link based on the window.location.href

### Issue tracking

https://govukverify.atlassian.net/browse/PYIC-6943

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

### CODE REVIEW

File: journey-map/public/index.mjs [hunk 1]
```diff
@@ -67,6 +67,8 @@ const stateSearch = document.getElementById('state-search');
 const journeyDesc = document.getElementById('journeyDesc');
 const journeyName = document.getElementById('journeyName');
 
+document.getElementById('header-title').href = window.location.href.split('?')[0];
```
Comment (Joe-Edwards-GDS) [inline journey-map/public/index.mjs]: I'd prefer this to match the mechanism used in our other links - we can leave the href fixed as `/`, but override the click behaviour.

Perhaps something like:
```
link.onclick = async (e) => {
  e.preventDefault();
  await switchJourney(DEFAULT_JOURNEY_TYPE);
};
```

I've also been trying to put all the selectors in the block above, and any 'initialisation logic' in the `initialize` method at the bottom - that makes it easier to keep track of all the bits and pieces.

**Review by Joe-Edwards-GDS — COMMENTED** [review]
Thanks for spotting!

File: journey-map/public/index.mjs (hunk 1 above)
Comment (sam803) [reply journey-map/public/index.mjs]: @Joe-Edwards-GDS thanks, done :)

**Review by Joe-Edwards-GDS — APPROVED** [review]
Thanks!

**Pull request merged**